### PR TITLE
Batch state updates #1

### PR DESCRIPTION
**Pull request by taoalpha**
https://github.com/polymorpher/one-wallet/issues/206

### CODE REVIEW

**Comment by polymorpher** [comment]
looks good, have you had a chance to look at https://github.com/polymorpher/one-wallet/pull/199#issuecomment-991871506 and see if the fundamental issues with SET_KNOWN_ADDRESS can be fixed? It shouldn't be called every time the page loads.

**Comment by taoalpha** [comment]
will look into that later

**Comment by polymorpher** [comment]
Redux batch seems to be designed to minimize render updates to React. (https://react-redux.js.org/api/batch). The changes you made fixed that and should speed up things a lot for rendering. Another concern discussed in the comment linked above is related to cross-tab propagation issues - whenever a state update is triggered, a cross-tab message is sent to other tabs and other tabs replaces the entire Redux store's state with the content of the message. If the state updates are too frequent, other tabs are overwhelmed with updates and may crash. Batching does not seem to reduce the number of state updates.

File: code/client/src/components/AddressInput.jsx [hunk 1]
```diff
@@ -163,39 +163,50 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
...
-
-      unlabelledWalletAddress.forEach((w) => {
-        dispatch(globalActions.setKnownAddress({
-          ...knownAddresses[w],
-          label: w.name,
-        }))
-      })
-
-      await Promise.all(knownAddressesWithoutDomain.map(async (knownAddress) => {
+      const domainWalletAddresses = await Promise.all(knownAddressesWithoutDomain.map(async (knownAddress) => {
```
Comment (polymorpher) [inline code/client/src/components/AddressInput.jsx:166]: There could be a long delay in resolving this. We shouldn't wait for this to finish before updating the rest

**Comment by polymorpher** [comment]
Other parts look good. We can merge this first (after the commented issue is fixed) and deal with cross-tab issue in separate PR

File: code/client/src/components/AddressInput.jsx (hunk 1 above)
Comment (taoalpha) [reply code/client/src/components/AddressInput.jsx]: Done

**Pull request merged**